collie/metrics.py

```diff
@@ -492,17 +494,20 @@ def explicit_evaluate_in_batches(
 
 
 def _get_evaluate_in_batches_device(model: BasePipeline):
-    device = getattr(model, 'device') or ('cuda:0' if torch.cuda.is_available() else 'cpu')
+    device = getattr(model, 'device', None)
 
-    if torch.cuda.is_available() and getattr(model, 'device') == 'cpu':
+    if torch.cuda.is_available() and str(device) == 'cpu':
         warnings.warn('CUDA available but model device is set to CPU - is this desired?')
 
+    if device is None:
+        device = 'cuda:0' if torch.cuda.is_available() else 'cpu'
```

Comment:
It's only going to be put on the GPU if `torch.cuda.is_available()`. Would you like a warning raised if `device is None and torch.cuda.is_available()`?  Similar to L499-500 above.

Reply:
Maybe? It is too overkill to raise a warning like: 

```
WARNING: ``model.device`` attribute is ``None``. Since GPU is available, putting model on GPU.
```

It might be too overkill... if you think it is, feel free to just ignore.

Reply:
well, we may as well add it since it isn't necessarily the expected outcome in the very rare case of model.device being none